pkg/oci/containerd.go

```diff
@@ -536,20 +541,34 @@ func AddMirrorConfiguration(ctx context.Context, fs afero.Fs, configPath string,
 		capabilities = append(capabilities, "resolve")
 	}
 	for _, registryURL := range registryURLs {
-		// Need a special case for Docker Hub as docker.io is just an alias.
 		server := registryURL.String()
 		if registryURL.String() == "https://docker.io" {
 			server = "https://registry-1.docker.io"
 		}
-		hostConfigs := map[string]hostConfig{}
-		for _, u := range mirrorURLs {
-			hostConfigs[u.String()] = hostConfig{Capabilities: capabilities}
-		}
-		cfg := hostFile{
+		hf := hostFile{
 			Server:      server,
-			HostConfigs: hostConfigs,
+			HostConfigs: map[string]hostConfig{},
+		}
+		if appendToBackup {
+			fp := path.Join(configPath, backupDir, registryURL.Host, "hosts.toml")
+			b, err := afero.ReadFile(fs, fp)
+			if err != nil && !errors.Is(err, afero.ErrFileNotFound) {
+				return err
+			}
+			if err == nil {
+				hf = hostFile{}
+				err := toml.Unmarshal(b, &hf)
+				if err != nil {
+					return err
+				}
+				log.Info("appending to existing Containerd mirror configuration", "registry", registryURL.String(), "path", fp)
+			}
+		}
```

Comment:
I think I would have formulated this as:
```go
hf := hostFile{
	Server:      server,
	HostConfigs: map[string]hostConfig{},
}
backupUsed := false
if appendToBackup {
	var err
	fp := path.Join(configPath, backupDir, registryURL.Host, "hosts.toml")
	hf, backupUsed, err = loadBackupHostFile(fs, fp)
	if err != nil {
		return err
	}
}
```
Using `backupUsed` we can then generate separate log messages for the separate cases rather than generating an extra log message in the append case.

This leads to three improvements (imo):
- it is clear we read backups from the helper name and the parameter can retain the name`appendMirrors` instead of switching to `appnedToBackup`, and
- it is easier to see that we are replacing `hf` in the append case, and
- denser log output

Reply:
I totally agree with you.

I had the intention of refactoring this code at a later time when implementing the CRIO version. After some trouble understanding my own code I chose to refactor it now. Added your suggestion to reduce the amount of logs.